packages/api/tests/entities/Module.test.ts

```diff
@@ -83,31 +104,67 @@ describe('Module entity', () => {
         previousId: modulePrev.id,
         pathId: path2.id
       };
+
       const updated = await TestClient.updateModule(update);
 
       expect(updated.id).toBeDefined();
-      expect(updated.name).toEqual(update.name);
-      expect(updated.icon).toEqual(update.icon);
-      expect(updated.type).toEqual(update.type);
-      expect(updated.pathId).toEqual(path2.id);
-      expect(updated.previousId).toEqual(modulePrev.id);
-
+      expect(updated).toMatchObject(update);
     });
   });
 
   describe('Mutation: deleteModule', () => {
     beforeEach(async () => {
       await setup();
-      path = await TestClient.createPath(pathInput, characterInput);
+      path = await TestClient.createPath(pathInput);
     });
 
     it('should delete a module successfully', async () => {
       expect.assertions(1);
-      const module = await TestClient.createModule({ ...moduleInput, pathId: path.id });
+      const module = await TestClient.createModule({
+        ...moduleInput,
+        pathId: path.id
+      });
       await TestClient.deleteModule(module.id);
       const modules = await TestClient.modules();
       expect(modules.length).toBe(0);
+    });
+  });
 
+  describe('Mutation: Join Module', () => {
+    beforeEach(async () => {
+      await setup();
+      path = await TestClient.createPath(pathInput);
+    });
+
+    it('join a module successfully', async () => {
+      expect.assertions(1);
+
+      const module = await TestClient.createModule({
+        ...moduleInput,
+        pathId: path.id
+      });
+
+      await TestClient.createUser();
+      const res = await TestClient.joinModule(module.id);
+      await expect(res).toBe(true);
+    });
+
+    it('should not allow to join a module twice', async () => {
+      expect.assertions(1);
+
+      const module = await TestClient.createModule({
+        ...moduleInput,
+        pathId: path.id
+      });
+
+      await TestClient.createUser();
```

Comment:
What is this for?

Reply:
Sorry. Thought I had to create a user before I can test the joinModule function. This has been removed. Thanks @quimgv .